### PR TITLE
Hashing password

### PR DESCRIPTION
**Pull request by romanpovol**

### CODE REVIEW

File: Client/app/src/main/java/ru/hse/client/auth/AuthUtility.kt [hunk 1]
```diff
@@ -120,3 +125,26 @@ fun Context.hideKeyboard(view: View) {
     inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
 }
 
+private const val ALGORITHM = "PBKDF2WithHmacSHA512"
+private const val ITERATIONS = 12000
+private const val KEY_LENGTH = 256
+private const val SECRET = "BOBER9107814265"
```
Comment (Redvin-dt) [inline Client/app/src/main/java/ru/hse/client/auth/AuthUtility.kt]: Should be moved to some config like application.properties on server, and then use it as in RequestFileValidationService

File: Client/app/src/main/java/ru/hse/client/auth/Requests.kt [hunk 2]
```diff
@@ -103,8 +103,8 @@ fun tryToLogInUser(
                 val responseBody: ByteString? = response.body?.byteString()
                 val user_: EntitiesProto.UserModel =
                     EntitiesProto.UserModel.parseFrom(responseBody?.toByteArray())
-                if (user_.hasPasswordHash()) {
-                    if (user_.passwordHash == password) {
+                if (user_.hasPasswordHash() && user_.hasPasswordSalt()) {
+                    if (user_.passwordHash == generateHash(password, user_.passwordSalt)) {
```
Comment (Redvin-dt) [inline Client/app/src/main/java/ru/hse/client/auth/Requests.kt]: Remove salt from hashing in client

File: Client/app/src/main/java/ru/hse/client/auth/AuthUtility.kt (hunk 1 above)
Comment (Redvin-dt) [inline Client/app/src/main/java/ru/hse/client/auth/AuthUtility.kt]: this field should be private for this repository

File: proto/Entities.proto [hunk 3]
```diff
@@ -41,10 +41,11 @@ message UserModel {
   optional string login = 2;
   optional string passwordHash = 3;
   optional string email = 4;
+  optional string passwordSalt = 5;
```
Comment (Redvin-dt) [inline proto/Entities.proto]: remove

File: server/src/main/java/ru/hse/server/utils/ProtoSerializer.java [hunk 4]
```diff
@@ -30,6 +31,7 @@ static public UserModel getUserInfo(User user) {
                 .setLogin(user.getUserLogin())
                 .setEmail(user.getUserEmail())
                 .setPasswordHash(user.getPasswordHash())
+                .setPasswordSalt(user.getPasswordSalt())
```
Comment (Redvin-dt) [inline server/src/main/java/ru/hse/server/utils/ProtoSerializer.java]: remove

File: server/src/main/java/ru/hse/server/utils/ProtoSerializer.java [hunk 5]
```diff
@@ -78,6 +80,7 @@ static public UserModel getProtoFromUser(User user) {
...
```
Comment (Redvin-dt) [inline server/src/main/java/ru/hse/server/utils/ProtoSerializer.java]: remove

**Review by Redvin-dt — CHANGES_REQUESTED** [review]

**Comment by Redvin-dt** [comment]
Working with tokens:
1) Now you can see white list in application.properties, there we store endpoint that can be used without authorisation
2) To use other endpoint you need to set header "Authorization" with value "Bearer <token>", otherwise you will get status 403 forbidden or 401 unauthorized
3) Token your get on endpoints login and registration, there is new field 'accessToken' in UserModel, that will fills only in that endpoints
4) Example how to use it, you may check on Request.kt:177

File: server/build.gradle.kts [hunk 6]
```diff
@@ -41,6 +41,7 @@ dependencies {
     testImplementation("org.springframework.boot:spring-boot-starter-test")
     implementation("org.springframework.boot:spring-boot-starter-security")
     testImplementation("org.springframework.security:spring-security-test")
+    compileOnly("org.springframework.boot:spring-boot-configuration-processor")
```
Comment (devcrocod) [inline server/build.gradle.kts]: I guess compileOnly is deprecated

File: server/src/main/java/ru/hse/server/filter/JWTFilter.java [hunk 7]
```diff
@@ -22,7 +22,7 @@
 public class JWTFilter extends OncePerRequestFilter {
     private static final String AUTHORIZATION_HEADER = "Authorization";
 
-    private final JWTProviderService jwtProviderService;
+    private final JWTProvider jwtProviderService;
```
Comment (devcrocod) [inline server/src/main/java/ru/hse/server/filter/JWTFilter.java:25]: How this service initializes?

File: server/src/main/java/ru/hse/server/service/AuthService.java [hunk 8]
```diff
@@ -27,8 +29,8 @@ public EntitiesProto.UserModel login(@Nonnull EntitiesProto.UserModel userModel)
             throw new AuthException("no such user");
         }
 
-        if (user.getPasswordHash().equals(userModel.getPasswordHash())) { // TODO: add hashing for password
-            return ProtoSerializer.getUserInfo(user).toBuilder().setAccessToken(jwtProviderService.generateAccessToken(user)).build();
+        if (passwordEncoderService.matchPassword(userModel.getPasswordHash(), user.getPasswordHash())) {
+            return ProtoSerializer.getUserInfo(user).toBuilder().setAccessToken(jwtProvider.generateAccessToken(user)).build();
```
Comment (devcrocod) [inline server/src/main/java/ru/hse/server/service/AuthService.java]: with chain calls, use a new line for each function call

**Review by devcrocod — APPROVED** [review]

File: server/src/main/java/ru/hse/server/filter/JWTFilter.java (hunk 7 above)
Comment (Redvin-dt) [reply server/src/main/java/ru/hse/server/filter/JWTFilter.java:25]: With annotation RequiredArgsConstructor, it create constructor that expect JWTProvider

File: Client/app/src/main/assets/app.properties
Comment (YuriiFilatov) [inline Client/app/src/main/assets/app.properties:1]: Secret info must be deleted

**Review by YuriiFilatov — APPROVED** [review]

**Review by Redvin-dt — APPROVED** [review]

**Pull request merged**